### PR TITLE
Implement CatchAll and StopCatch in signal_unix.go

### PR DESCRIPTION
**Pull request by dfr**
[NO NEW TESTS NEEDED]

Signed-off-by: Doug Rabson <dfr@rabson.org>

#### Does this PR introduce a user-facing change?

No

```release-note
None
```

### CODE REVIEW

File: pkg/signal/signal_unix.go [hunk 1]
```diff
@@ -91,10 +92,15 @@ var SignalMap = map[string]syscall.Signal{
 
 // CatchAll catches all signals and relays them to the specified channel.
 func CatchAll(sigc chan os.Signal) {
-	panic("Unsupported on non-linux platforms")
+	handledSigs := make([]os.Signal, 0, len(SignalMap))
+	for _, s := range SignalMap {
+		handledSigs = append(handledSigs, s)
+	}
+	signal.Notify(sigc, handledSigs...)
```
Comment (Luap99) [inline pkg/signal/signal_unix.go:99]: Maybe I am missing something but this only uses go standard library functionality so it should work on all platforms.
I think moving this into signal_common.go makes more sense.

Comment (dfr) [reply pkg/signal/signal_unix.go:99]: Good point - I'll move it

**Review by vrothberg — COMMENTED** [review]
Changes LGTM but the commit message lacks clarification. What problem does the commit address and why are no tests needed?

**Comment by dfr** [comment]
This is part of a set of changes to port podman to the FreeBSD platform. The pkg/signal parts are needed to enable ABI mode on FreeBSD. No tests are needed here because it should be a functional no-op for linux.

I can update the commit message to make this clearer or alternatively, I can stack it with the commit which enables ABI mode instead of putting that in its own PR. Thoughts?

**Comment by dfr** [comment]
I've updated the commit message and also fixed the podman-remote-windows build break.

**Review by Luap99 — APPROVED** [review]
/lgtm
/hold

**Review by vrothberg — COMMENTED** [review]
LGTM, thank you!

**Pull request merged**